spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/TypeToSparkType.java

```diff
@@ -69,9 +70,24 @@ public DataType struct(Types.StructType struct, List<DataType> fieldResults) {
       if (field.doc() != null) {
         sparkField = sparkField.withComment(field.doc());
       }
+      if (field.writeDefault() != null) {
+        // Convert Iceberg default value to Spark SQL string representation. Spark stores default
+        // values as SQL strings in column metadata. The .sql() method formats literals correctly
+        // for each type
```

Comment:
If we just remove the comment below for the initial default case, I'd just have a more concise comment that just spans both, above this if block. "Convert both write and initial default values to Spark SQL string literal representations on the StructField metadata"